### ISSUE TITLE
Installation on Octave+Mac OSX fails

### ISSUE THREAD

**Issue by nils-werner** [ltfat/ltfat#22]
I have Octave from Homebrew and tried to install ltfat using the builin `pkg install` command:

    [user@mac ~]$ octave
    GNU Octave, version 3.8.1
    Copyright (C) 2014 John W. Eaton and others.
    This is free software; see the source code for copying conditions.
    There is ABSOLUTELY NO WARRANTY; not even for MERCHANTABILITY or
    FITNESS FOR A PARTICULAR PURPOSE.  For details, type 'warranty'.

    Octave was configured for "x86_64-apple-darwin14.0.0".

    Additional information about Octave is available at http://www.octave.org.

    Please contribute if you find this software useful.
    For more information, visit http://www.octave.org/get-involved.html

    Read http://www.octave.org/bugs.html to learn how to submit bug reports.
    For information about changes from previous versions, type 'news'.

    octave:1> pkg install -forge ltfat
    In file included from dct_ci.c:6:
    ./dct.c:36:37: warning: implicit conversion from enumeration type
          'const dct_kind' to different enumeration type 'fftw_r2r_kind' (aka 'enum
          fftw_r2r_kind_do_not_use_me') [-Wenum-conversion]
        LTFAT_FFTW(r2r_kind) kindFftw = kind;
                             ~~~~~~~~   ^~~~
    In file included from dct_ci.c:10:
    ./dct.c:36:37: warning: implicit conversion from enumeration type
          'const dct_kind' to different enumeration type 'fftw_r2r_kind' (aka 'enum
          fftw_r2r_kind_do_not_use_me') [-Wenum-conversion]
        LTFAT_FFTW(r2r_kind) kindFftw = kind;
                             ~~~~~~~~   ^~~~
    2 warnings generated.
    In file included from dst_ci.c:6:
    ./dst.c:36:37: warning: implicit conversion from enumeration type
          'const dst_kind' to different enumeration type 'fftw_r2r_kind' (aka 'enum
          fftw_r2r_kind_do_not_use_me') [-Wenum-conversion]
        LTFAT_FFTW(r2r_kind) kindFftw = kind;
                             ~~~~~~~~   ^~~~
    In file included from dst_ci.c:10:
    ./dst.c:36:37: warning: implicit conversion from enumeration type
          'const dst_kind' to different enumeration type 'fftw_r2r_kind' (aka 'enum
          fftw_r2r_kind_do_not_use_me') [-Wenum-conversion]
        LTFAT_FFTW(r2r_kind) kindFftw = kind;
                             ~~~~~~~~   ^~~~
    2 warnings generated.
    ar: creating archive libltfat.a
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libltfat.a(dspread.o) has no symbols
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libltfat.a(dspread.o) has no symbols
    In file included from dct_ci.c:6:
    ./dct.c:36:37: warning: implicit conversion from enumeration type
          'const dct_kind' to different enumeration type 'fftwf_r2r_kind' (aka 'enum
          fftw_r2r_kind_do_not_use_me') [-Wenum-conversion]
        LTFAT_FFTW(r2r_kind) kindFftw = kind;
                             ~~~~~~~~   ^~~~
    In file included from dct_ci.c:10:
    ./dct.c:36:37: warning: implicit conversion from enumeration type
          'const dct_kind' to different enumeration type 'fftwf_r2r_kind' (aka 'enum
          fftw_r2r_kind_do_not_use_me') [-Wenum-conversion]
        LTFAT_FFTW(r2r_kind) kindFftw = kind;
                             ~~~~~~~~   ^~~~
    2 warnings generated.
    In file included from dst_ci.c:6:
    ./dst.c:36:37: warning: implicit conversion from enumeration type
          'const dst_kind' to different enumeration type 'fftwf_r2r_kind' (aka 'enum
          fftw_r2r_kind_do_not_use_me') [-Wenum-conversion]
        LTFAT_FFTW(r2r_kind) kindFftw = kind;
                             ~~~~~~~~   ^~~~
    In file included from dst_ci.c:10:
    ./dst.c:36:37: warning: implicit conversion from enumeration type
          'const dst_kind' to different enumeration type 'fftwf_r2r_kind' (aka 'enum
          fftw_r2r_kind_do_not_use_me') [-Wenum-conversion]
        LTFAT_FFTW(r2r_kind) kindFftw = kind;
                             ~~~~~~~~   ^~~~
    2 warnings generated.
    ar: creating archive libltfatf.a
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libltfatf.a(sspread.o) has no symbols
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libltfatf.a(sspread.o) has no symbols
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    mkoctfile: stripping disabled on this platform
    xargs: illegal option -- i
    usage: xargs [-0opt] [-E eofstr] [-I replstr [-R replacements]] [-J replstr]
                 [-L number] [-n number [-x]] [-P maxprocs] [-s size]
                 [utility [argument ...]]
    make: *** [oct] Error 1
    /Applications/Xcode.app/Contents/Developer/usr/bin/make -f Makefile_unix
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c dgt.c -o ddgt.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c dgt_fac.c -o ddgt_fac.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c dgt_fb.c -o ddgt_fb.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c dgt_multi.c -o ddgt_multi.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c dgt_ola.c -o ddgt_ola.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c dgt_shear.c -o ddgt_shear.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c dgt_walnut.c -o ddgt_walnut.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c dgtreal_fac.c -o ddgtreal_fac.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c dwilt.c -o ddwilt.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c idwilt.c -o didwilt.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c wmdct.c -o dwmdct.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c iwmdct.c -o diwmdct.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c filterbank.c -o dfilterbank.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c ifilterbank.c -o difilterbank.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c heapint.c -o dheapint.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c idgt.c -o didgt.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c idgt_fac.c -o didgt_fac.occ -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c idgt_fb.c -o didgt_fb.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c iwfac.c -o diwfac.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c pfilt.c -o dpfilt.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c reassign.c -o dreassign.occ -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c spread.c -o dspread.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c wfac.c -o dwfac.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c windows.c -o dwindows.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c winmanip.c -o dwinmanip.occ -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c ltfat_complexindependent.c -o dltfat_complexindependent.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c dgt_shearola.c -o ddgt_shearola.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c dct_ci.c -o ddct_ci.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c dst_ci.c -o ddst_ci.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c fftreal.c -o dfftreal.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c ltfat_blaslapack.c -o dltfat_blaslapack.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c gabdual_fac.c -o dgabdual_fac.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_DOUBLE  -c gabtight_fac.c -o dgabtight_fac.o
    bl.c -o dltfat_complexindependent_bl.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -c c-safe-memalloc.c
    cc -O3 -fPIC -std=c99 -Ithirdparty  -c integer_manip.c
    ar rvu libltfat.a ddgt.o ddgt_fac.o ddgt_fb.o ddgt_multi.o ddgt_ola.o ddgt_shear.o ddgt_walnut.o ddgtreal_fac.o ddwilt.o didwilt.o dwmdct.o diwmdct.o dfilterbank.o difilterbank.o dheapint.o didgt.o didgt_fac.o didgt_fb.o diwfac.o dpfilt.o dreassign.o dspread.o dwfac.o dwindows.o dwinmanip.o dltfat_complexindependent.o ddgt_shearola.o ddct_ci.o ddst_ci.o dfftreal.o dltfat_blaslapack.o dgabdual_fac.o dgabtight_fac.o dltfat_complexindependent_bl.o c-safe-memalloc.o integer_manip.o 
    a - ddgt.o
    a - ddgt_fac.o
    a - ddgt_fb.o
    a - ddgt_multi.o
    a - ddgt_ola.o
    a - ddgt_shear.o
    a - ddgt_walnut.o
    a - ddgtreal_fac.o
    a - ddwilt.o
    a - didwilt.o
    a - dwmdct.o
    a - diwmdct.o
    a - dfilterbank.o
    a - difilterbank.o
    a - dheapint.o
    a - didgt.o
    a - didgt_fac.o
    a - didgt_fb.o
    a - diwfac.o
    a - dpfilt.o
    a - dreassign.o
    a - dspread.o
    a - dwfac.o
    a - dwindows.o
    a - dwinmanip.o
    a - dltfat_complexindependent.o
    a - ddgt_shearola.o
    a - ddct_ci.o
    a - ddst_ci.o
    a - dfftreal.o
    a - dltfat_blaslapack.o
    a - dgabdual_fac.o
    a - dgabtight_fac.o
    a - dltfat_complexindependent_bl.o
    a - c-safe-memalloc.o
    a - integer_manip.o
    ranlib libltfat.a
    cp -f libltfat.a ../lib
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c dgt.c -o sdgt.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c dgt_fac.c -o sdgt_fac.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c dgt_fb.c -o sdgt_fb.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c dgt_multi.c -o sdgt_multi.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c dgt_ola.c -o sdgt_ola.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c dgt_shear.c -o sdgt_shear.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c dgt_walnut.c -o sdgt_walnut.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c dgtreal_fac.c -o sdgtreal_fac.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c dwilt.c -o sdwilt.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c idwilt.c -o sidwilt.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c wmdct.c -o swmdct.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c iwmdct.c -o siwmdct.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c filterbank.c -o sfilterbank.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c ifilterbank.c -o sifilterbank.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c heapint.c -o sheapint.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c idgt.c -o sidgt.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c idgt_fac.c -o sidgt_fac.occ -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c idgt_fb.c -o sidgt_fb.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c iwfac.c -o siwfac.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c pfilt.c -o spfilt.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c reassign.c -o sreassign.occ -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c spread.c -o sspread.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c wfac.c -o swfac.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c windows.c -o swindows.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c winmanip.c -o swinmanip.occ -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c ltfat_complexindependent.c -o sltfat_complexindependent.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c dgt_shearola.c -o sdgt_shearola.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c dct_ci.c -o sdct_ci.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c dst_ci.c -o sdst_ci.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c fftreal.c -o sfftreal.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c ltfat_blaslapack.c -o sltfat_blaslapack.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c gabdual_fac.c -o sgabdual_fac.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c gabtight_fac.c -o sgabtight_fac.o
    cc -O3 -fPIC -std=c99 -Ithirdparty  -DLTFAT_SINGLE  -c ltfat_complexindependent_bl.c -o sltfat_complexindependent_bl.o
    ar rvu libltfatf.a sdgt.o sdgt_fac.o sdgt_fb.o sdgt_multi.o sdgt_ola.o sdgt_shear.o sdgt_walnut.o sdgtreal_fac.o sdwilt.o sidwilt.o swmdct.o siwmdct.o sfilterbank.o sifilterbank.o sheapint.o sidgt.o sidgt_fac.o sidgt_fb.o siwfac.o spfilt.o sreassign.o sspread.o swfac.o swindows.o swinmanip.o sltfat_complexindependent.o sdgt_shearola.o sdct_ci.o sdst_ci.o sfftreal.o sltfat_blaslapack.o sgabdual_fac.o sgabtight_fac.o sltfat_complexindependent_bl.o 
    a - sdgt.o
    a - sdgt_fac.o
    a - sdgt_fb.o
    a - sdgt_multi.o
    a - sdgt_ola.o
    a - sdgt_shear.o
    a - sdgt_walnut.o
    a - sdgtreal_fac.o
    a - sdwilt.o
    a - sidwilt.o
    a - swmdct.o
    a - siwmdct.o
    a - sfilterbank.o
    a - sifilterbank.o
    a - sheapint.o
    a - sidgt.o
    a - sidgt_fac.o
    a - sidgt_fb.o
    a - siwfac.o
    a - spfilt.o
    a - sreassign.o
    a - sspread.o
    a - swfac.o
    a - swindows.o
    a - swinmanip.o
    a - sltfat_complexindependent.o
    a - sdgt_shearola.o
    a - sdct_ci.o
    a - sdst_ci.o
    a - sfftreal.o
    a - sltfat_blaslapack.o
    a - sgabdual_fac.o
    a - sgabtight_fac.o
    a - sltfat_complexindependent_bl.o
    ranlib libltfatf.a
    cp -f libltfatf.a ../lib
    /Applications/Xcode.app/Contents/Developer/usr/bin/make -C ../oct -f Makefile_unix
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_atrousfilterbank_td.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_cellcoef2tf.cc/usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_chirpzt.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_col2diag.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_dct.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_dgt_ola.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_dgtreal_ola.cc/usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_dst.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_dwilt.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_dwiltiii.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_fftreal.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_filterbank_fft.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_filterbank_fftbl.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_filterbank_td.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_gabdual_long.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_gabreassign.cc/usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_gabtight_long.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_gga.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_heapint.cc
    ank_td.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_idwilt.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_idwiltiii.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_ifftreal.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_ifilterbank_fft.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_ifilterbank_fftbl.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_ifilterbank_td.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_isepdgt.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_isepdgtreal.cc/usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_iwfac.cc
    y -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_nonsepdgt_multi.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_nonsepdgt_shear.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_nonsepwin2multi.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_pchirp.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_pgauss.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_sepdgt.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_sepdgtreal.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_ufilterbank_fft.cc
    /usr/local/Cellar/octave/3.8.1_1/bin/mkoctfile-3.8.1 -strip -Wall -I../thirdparty -I. -I../src -I../src/thirdparty -L../lib -lltfat -lltfatf comp_wfac.cc
    mv ../oct/*.oct .
    ls -1 *.oct | cut -d . -f 1 | xargs -i rm -f ../inst/comp/{}.m

    pkg: error running `make' for the ltfat package.
    error: called from 'configure_make' in file /usr/local/Cellar/octave/3.8.1_1/share/octave/3.8.1/m/pkg/private/configure_make.m near line 82, column 9
    error: called from:
    error:   /usr/local/Cellar/octave/3.8.1_1/share/octave/3.8.1/m/pkg/private/install.m at line 199, column 5
    error:   /usr/local/Cellar/octave/3.8.1_1/share/octave/3.8.1/m/pkg/pkg.m at line 394, column 9

**Comment by jvnv** [comment]
Dear Nils Werner,

Thanks for opening this issue. The problem you encountered is unfortunately a familiar problem of the combination LTFAT, Octave and Mac OS X, and isn't solved yet. 
For now, I would like to recommend you to download LTFAT manually if you are interested in using it. LTFAT can be downloaded from the link below.

Link: http://ltfat.sourceforge.net

After downloading the binary or developers version of LTFAT, it should work without any problem with Octave on Mac OS X.

Best regards,

Jordy

**Comment by susnak** [comment]
Hello, 
the error happends at the very end of the build process while trying to delete duplicate m-files which were compiled as oct files.
The exact reason this fails is here:
ls -1 *.oct | cut -d . -f 1 | xargs -i rm -f ../inst/comp/{}.m
xargs: illegal option -- i
Apparently we are using -i flag, which is not supported on Mac.
This can be easily fixed by downloading the package ltfat-2.0.1.tar.gz from the Octave Forge and changing line 19 in src/Makefile.in to
ls -1 *.oct | cut -d . -f 1 | xargs -I {} rm -f ../inst/comp/{}.m
From Octave, you can then run 
pkg install ltfat-2.0.1.tar.gz
(provided you are in the correct directory)

Please let me know whether this works. I will change that in the future release.

The warnings you get come from clang. I will have to have a look at that. They should not be critical.

**Comment by psoenderg** [comment]
It would probably be safer rewriting the line using a small for-loop, then we would not have to rely on xargs behaving in the same manner on Linux and Mac.

**Comment by nils-werner** [comment]
You can simply replace `-i` with `-I{}`.

In fact, you should as `-i` hasn't been in the POSIX standard since 2004 (only in the one from 1997) and has been marked as deprecated on almost every man page.

**Comment by psoenderg** [comment]
But my fingers have been typing "-i" for 15 years now, that is going to be hard to change :-) I guess it is about time I start updating my command line arguments. Good to know that -I{} also works with Mac, I don't think we have a Mac computer to test on. 

Cheers,
Peter.

**Comment by nils-werner** [comment]
I am confused... Why is there a 2.0.1 on octave forge, and ltfat and this repo are at 2.0.0?

**Comment by susnak** [comment]
Well, after releasing 2.0.0 I found out I need to include tests in the Octave package so I did a new release.
The numbers will again be the same in the next release ( 2.0.2).

**Comment by nils-werner** [comment]
And where is 2.0.1 now? Why isn't it here?

**Comment by susnak** [comment]
How did you find out this repo is at 2.0.0?
ltfat_version contains 2.0.1
changeLog is prepared for 2.0.2
NEWS says 2.0.1
The only place where we still have 2.0.0 is src/configure.ac
Which is updated automatically when we do a release.

We do not have the code organized as an Octave package in our repository. The Octave package is extracted from the repository by a release script. The same script creates Matlab release packages for Windows and Unix systems which has the files organized differently.
The script removes the tests. I was asked to include the tests so the ltfat Debian package build system has something to run after the build is complete. They however would not accept any code which was not officially released on Octave Forge so I had to do a new one.

To summarize:
2.0.1 is here in the repository as all other versions ever released. The release packages are at sourceforge (for now) and the Octave package is at Octave Forge.

**Comment by nils-werner** [comment]
Extracting the tgz files and looking at the directory structure looked much more like 2.0.0 so I assumed it was the case.

Not being able to find a package/release script I assumed this was the final package structure. Can you point me to the release script so I can test if your fix works?

**Comment by susnak** [comment]
Would downloading ltfat-2.0.1.tar.gz directly be enough? It can be found here

https://sourceforge.net/projects/octave/files/Octave%20Forge%20Packages/Individual%20Package%20Releases/

This is the place it is downloaded from when you run
pkg install ltfat -forge

Then you can change src/Makefile.in and install (without -forge)
pkg install ltfat-2.0.1.tar.gz

**Comment by nils-werner** [comment]
No, I would like to run the code from here, fix bugs etc. :-)

**Comment by susnak** [comment]
For that you do not need to run the release script. You can run ltfat directly from the clonned repository. 
See ltfatstart and ltfatmex.
Fixing bugs on the release package is actually a bad idea as the code itself (mainly m-function headers) is processed heavily by the release script. 

Of course for bugs like this one you actualy need to create a package.
We use a Python script mat2doc https://sourceforge.net/projects/mat2doc/ using this call
mat2doc.py [pathtorepo] mat --script=release.py --octpkg --unix
It however requires some setting up (see conf.py in ltfat/mat2doc and in ltfat/mat2doc/mat )

**Comment by psoenderg** [comment]
Bug fixed, and fix commited.
